### ISSUE TITLE
Postgres reports long running "<idle> in transaction", starting after the first database write

### ISSUE THREAD

**Issue by janwijbrand** [zodb/relstorage#7]
Hi,

Recently we realized Relstorage seems to start and never close a database transaction after the first write. This database connection is reported to be ```<IDLE> in transaction```. See the query below.

Note: this idling transaction is only started after adding an object to a "container" - in our testing setup, it was adding an ```object()``` to the Zope application root. Storing an attribute on the app root, seems to not result in leaving an idleing transaction.

We suspect that this idling transaction keeps postgres from successfully running its auto vacuuming task. 

Is this expected behaviour? Do I need to provide more details?

Kind regards, jw

```
postgres=# SELECT procpid, current_timestamp - xact_start AS xact_runtime, current_query
FROM pg_stat_activity ORDER BY xact_start;
 procpid |      xact_runtime      |                                 current_query                                 
---------+------------------------+-------------------------------------------------------------------------------
   25473 | 2 days 04:21:24.419343 | <IDLE> in transaction
   13115 | 1 day 16:33:19.276329  | <IDLE> in transaction
   32072 | 1 day 14:35:30.829614  | <IDLE> in transaction
    1577 | 00:00:00.631836        | <IDLE> in transaction
   26561 | 00:00:00               | SELECT procpid, current_timestamp - xact_start AS xact_runtime, current_query
                                  : FROM pg_stat_activity ORDER BY xact_start;
```

**Comment by do3cc** [comment]
https://code.djangoproject.com/ticket/9964#comment:34

**Comment by mauritsvanrees** [comment]
Also, if you are using RelStorage 1.5.0 or older, this might be a problem that was fixed in 1.5.1 (November 2011). From the changelog: "RelStorage was opening a test database connection (and was leaving it idle in a transaction with recent ZODB versions that support IMVCCStorage.) RelStorage no longer opens that test connection."

**Comment by hathawsh** [comment]
Closing due to no response in 2 years.
